### ISSUE TITLE
Table - Column filters are not keyboard focusable

### ISSUE THREAD

**Issue by mellis481** [ant-design/ant-design#37508]
### Reproduction link

[https://ant.design/components/table/#components-table-demo-head](https://ant.design/components/table/#components-table-demo-head)

### Steps to reproduce

Hit "tab" key, navigating through focusable table elements.

### What is expected?

Like sortable columns, column filter buttons should be keyboard focusable.

### What is actually happening?

Column filter buttons are not keyboard focusable (because they have a tabindex of -1).

| Environment | Info |
| --- | --- |
| antd | 4.23.1 |
| React | 17 |
| System | latest |
| Browser | chrome |

**Comment by mellis481** [comment]
Would a PR to fix this be accepted?

**Comment by mellis481** [comment]
@afc163 @zombieJ Would a PR to fix this be accepted?

**Comment by yoyo837** [comment]
Yes, I personally think. PR is welcome.

**Comment by Ke1sy** [comment]
@afc163 @zombieJ  I have investigated current issue and would like to propose the solution how to fix it. Please let me know if you accept such PR?

In previous PR (https://github.com/ant-design/ant-design/pull/39326) we made the filter buttons accessible by adding the tab-index="0" to them and moving the focus into dropdown, but it hasn't solved the issue with navigation inside the filter dropdown. 
The next issues still existed:

1. For filterMode="menu":
  - after focusing the menu inside the dropdown focus gets locked in menu list, in a result all other content is unreachable including "Reset" / "Confirm" buttons ("Tab" is not working to navigate to next focusable item).

2. For filterMode="tree" and any custom content inside like search:
- "Tab" closes the dropdown menu.

The main issue that causes the navigation issues inside the dropdown are connected with `rc-dropdown`. The binding of the 'Tab' key in `src\\hooks\\useAccessibility.ts` > `handleKeyDown` closes the dropdown instead of navigating inside it.

To fix the focus issues in "menu" and "tree" modes - we need to override the key bindings in antd in `onVisibleChange` function to ignore `newVisible` value that comes from rc-dropdown (`components\\table\\hooks\\useFilter\\FilterDropdown.tsx`).

After that new key bindings for dropdown should be added in this component. It includes:

- the open/close dropdown `onClick` and `onKeyDown` on filter btn;
- handling the `Escape` key (close dropdown);
- controlling of the focus on open/close (btn => menu / menu => btn);
- ref cannot be used to control the focus as several instances of dropdown-menus/trigger-btns can be used in same table and will have same ref. May be use unique id for each pair of trigger/dropdown-menu.

Also I reccomend to add `react-focus-lock` package in antd and wrap the dropdown content inside it to lock the focus (components\\table\\hooks\\useFilter\\FilterDropdown.tsx). This will solve the issue with calculating the last focusable element in the dropdown to move focus back to the top or going out of the dropdown with "Tab".

**Comment by afc163** [comment]
@Ke1sy I have same idea about importing `react-focus-lock` to antd. PR is welcome to resolve this issue!

https://github.com/ant-design/ant-design/issues/47916
